RecoLocalCalo/HGCalRecAlgos/src/HGCalImagingAlgo.cc

```diff
@@ -504,53 +512,54 @@ void HGCalImagingAlgo::shareEnergy(const std::vector<KDNode>& incluster,
   const unsigned iterMax = 50;
   double diff = std::numeric_limits<double>::max();
   const double stoppingTolerance = 1e-8;
-  const double toleranceScaling = std::pow(std::max(1.0,seeds.size()-1.0),2.0);
+  const auto numberOfSeeds = seeds.size();
+  auto toleranceScaling = numberOfSeeds > 2 ? (numberOfSeeds-1)*(numberOfSeeds-1) : 1;
   std::vector<Point> prevCentroids;
-  std::vector<double> frac(seeds.size()), dist2(seeds.size());
+  std::vector<double> frac(numberOfSeeds), dist2(numberOfSeeds);
   while( iter++ < iterMax && diff > stoppingTolerance*toleranceScaling ) {
     for( unsigned i = 0; i < incluster.size(); ++i ) {
       const Hexel& ihit = incluster[i].data;
       double fracTot(0.0);
-      for( unsigned j = 0; j < seeds.size(); ++j ) {
-	double fraction = 0.0;
-	double d2 = ( std::pow(ihit.x - centroids[j].x(),2.0) +
-	       std::pow(ihit.y - centroids[j].y(),2.0) +
-	       std::pow(ihit.z - centroids[j].z(),2.0)   )/sigma2;
-	dist2[j] = d2;
-	// now we set the fractions up based on hit type
-	if( i == seeds[j] ) { // this cluster's seed
-	  fraction = 1.0;
-	} else if( isaseed[i] ) {
-	  fraction = 0.0;
-	} else {
-	  fraction = energies[j]*std::exp( -0.5*d2 );
-	}
-	fracTot += fraction;
-	frac[j] = fraction;
+      for( unsigned j = 0; j < numberOfSeeds; ++j ) {
+    double fraction = 0.0;
+    double d2 = ( std::pow(ihit.x - centroids[j].x(),2.0) +
+           std::pow(ihit.y - centroids[j].y(),2.0) +
+           std::pow(ihit.z - centroids[j].z(),2.0)   )/sigma2;
+    dist2[j] = d2;
+    // now we set the fractions up based on hit type
+    if( i == seeds[j] ) { // this cluster's seed
+      fraction = 1.0;
+    } else if( isaseed[i] ) {
+      fraction = 0.0;
+    } else {
+      fraction = energies[j]*std::exp( -0.5*d2 );
+    }
+    fracTot += fraction;
+    frac[j] = fraction;
       }
       // now that we have calculated all fractions for all hits
       // assign the new fractions
-      for( unsigned j = 0; j < seeds.size(); ++j ) {
-	if( fracTot > minFracTot ||
-	    ( i == seeds[j] && fracTot > 0.0 ) ) {
-	  outclusters[j][i] = frac[j]/fracTot;
-	} else {
-	  outclusters[j][i] = 0.0;
-	}
+      for( unsigned j = 0; j < numberOfSeeds; ++j ) {
+    if( fracTot > minFracTot ||
+        ( i == seeds[j] && fracTot > 0.0 ) ) {
+      outclusters[j][i] = frac[j]/fracTot;
+    } else {
+      outclusters[j][i] = 0.0;
+    }
```

Comment:
and here 
https://github.com/felicepantaleo/cmssw/blob/4a4de46f104e5240ed02977bf2526175a894ed43/RecoLocalCalo/HGCalRecAlgos/src/HGCalImagingAlgo.cc#L542-L550